validator/sawtooth_validator/config/validator.py

```diff
@@ -0,0 +1,187 @@
+# Copyright 2017 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------------
+
+import collections
+import logging
+import os
+
+import toml
+
+from sawtooth_validator.exceptions import LocalConfigurationError
+from sawtooth_validator.config.path import get_default_path_config
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+def load_toml_validator_config(filename):
```

Comment:
IMO, the current method is better, as it keeps the data separate from the load mechanism.

from_file() with no arguments is singleton-like, which is extremely poor in that it provides a methodology to take control away from the top-level application -- creating technical debt to be incurred later.  Better to force passing the correct path as determined by the top-level application logic.

Since this module need not know about application-level logic, from_opts() functionality is best left elsewhere.

If putting merge in the ValidatorConfig class, one could see it acting like set's union() and acting on self and others but still returning a new config.